### PR TITLE
GaleShapley feature selection

### PR DESCRIPTION
**Pull request by julia-shenshina**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna-ts/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna-ts/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: etna/transforms/gale_shapley.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+            n_segments=len(segment_regressors_ranking),
+            n_regressors=len(regressor_segments_ranking),
+        )
+        last_step_regressors_number = self.top_k % len(segment_regressors_ranking)
+        for step in range(gale_shapley_steps_number):
+            matches = self._gale_shapley_iteration(
+                segment_regressors_ranking=segment_regressors_ranking,
+                regressor_segments_ranking=regressor_segments_ranking,
+            )
+            if step == gale_shapley_steps_number - 1:
```
[the remark this answers is not in the document]
Comment (julia-shenshina) [reply etna/transforms/gale_shapley.py:324]: ➕,  in progress

[the remark this answers is not in the document]
Comment (julia-shenshina) [reply etna/transforms/gale_shapley.py:352]: just to be sure, is this (`top_k==7`) your case? https://github.com/tinkoff-ai/etna-ts/pull/284/files#diff-27993ca4fa272893d5eb713f25f45aaae5f20097d766cf5b53a733e73fd1cb6dR585

File: etna/transforms/gale_shapley.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        Returns
+        -------
+        name: str
+            name of regressor
+        """
+        if self.last_candidate is None:
+            self.last_candidate = 0
+        else:
+            self.last_candidate += 1
+        return self.ranked_candidate[self.last_candidate]
```
[the remark this answers is not in the document]
Comment (julia-shenshina) [reply etna/transforms/gale_shapley.py:95]: fixed

**Comment by martins0n** [comment]
🥇 Code is readable!

File: etna/transforms/gale_shapley.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ama16) [reply etna/transforms/gale_shapley.py:352]: yes

**Review by martins0n — APPROVED** [review]

**Pull request merged**